splunk_add_on_ucc_framework/entity/interval_entity.py

```diff
@@ -0,0 +1,92 @@
+#
+# Copyright 2024 Splunk Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+from typing import Optional, Dict, Any
+
+from splunk_add_on_ucc_framework.entity.entity import Entity
+
+
+class IntervalEntity(Entity):
+    def short_form(self) -> Dict[str, Any]:
+        return dict(self)
+
+    def long_form(self) -> Dict[str, Any]:
+        definition = {
+            "type": "text",
+            "field": self["field"],
+            "label": self["label"],
+            "validators": [
+                {
+                    "type": "regex",
+                    "errorMsg": f"{self['label']} must be either a non-negative number or -1.",
+                    "pattern": r"^(?:-1|0(?:\.\d+)?|[1-9]\d*(?:\.\d+)?)$",
```

Comment:
This regex doesn't work with a value like `00123`. Users ideally don't put values like this, but it would be good if we can handle it.
Also, should we have any maximum value? The modinput should at least run once a year or once in six months, something like that?
We can use regex and range validators for this case.

Reply:
If there are no limits for intervals in Splunk, I don't think we should come up with some default value.
There are `min` and `max` fields in the options so developers can set it.

Reply:
I will add leading 0's to the regex.